source/common/common/key_value_store_base.cc

```diff
@@ -35,12 +37,20 @@ KeyValueStoreBase::KeyValueStoreBase(Event::Dispatcher& dispatcher,
     : max_entries_(max_entries), flush_timer_(dispatcher.createTimer([this, flush_interval]() {
         flush();
         flush_timer_->enableTimer(flush_interval);
-      })) {
+      })),
+      ttl_manager([this](const auto& expired) { ttlCallback(expired); }, dispatcher,
+                  dispatcher.timeSource()) {
   if (flush_interval.count() > 0) {
     flush_timer_->enableTimer(flush_interval);
   }
 }
 
+void KeyValueStoreBase::ttlCallback(const std::vector<std::string>& expired) {
+  for (const auto& key : expired) {
+    remove(key);
```

Comment:
This is inefficient if there's a large number of keys.  Perhaps I should implement a remove_batch(keys) function for the class

Reply:
That seems like a wise idea. I also wonder if there is an opportunity to store less state (by avoiding the need to replicate the key value) for the timeouts but given how the TTL manager works, that seems non-trivial. So `remove(vector)` seems like a fine idea.

Reply:
I actually just looked into it and QuicheLinkedHashMap doesn't support batch deletion... guess this will have to be good enough for now unless we want to change the store_ type

Reply:
It's true that linked hash map doesn't have a bulk delete, but the remove() method calls flush after each remove. So this current implementation of ttlCallback will result in N flushes if N keys are removed. It seems like it might be nice to avoid the extra flushes if we could. WDYT?